content/blog/tidymodels-2023-q3/index.Rmd

```diff
@@ -0,0 +1,175 @@
+
+```{r}
+#| message: FALSE
+library(tidymodels)
+library(tidyclust)
+
+data("ames", package = "modeldata")
+```
+
+## Workshops
+
+One of the biggest areas of work for our team this quarter was getting ready for this year's [posit::conf](https://posit.co/conference/). This year, two 1-day workshops were available; "Introduction to tidymodels" and "Advanced tidymodels". All the material can be found on our workshop website
+[workshops.tidymodels.org](https://workshops.tidymodels.org/), with this workshop being archived as [posit::conf 2023 workshop](https://workshops.tidymodels.org/archive/2023-09-posit-conf/).
+
+Unless otherwise noted (i.e. not an original creation and reused from another source), these educational materials are licensed under Creative Commons Attribution [CC BY-SA 4.0](https://creativecommons.org/licenses/by-sa/4.0/).
+
+## Tidyclust update
+
+The latest release of tidyclust featured a round of bug fixes, documentation improvements and quality-of-life improvements.
+
+This release adds 2 new engines to the `k_means()` model. [klaR](https://tidyclust.tidymodels.org/reference/details_k_means_klaR.html) to run K-Modes models and [clustMixType](https://tidyclust.tidymodels.org/reference/details_k_means_clustMixType.html) to run K-prototypes. K-Modes is the categorical analog to K-means, meaning that it is intended to be used on only categorical data, and K-prototypes is the more general method that works with categorical and numeric data at the same time.
+
+If we were to fit a K-means model to a mixed-type data set such `ames`, it would work but under the hood, it applied a dummy transformation on the categorical predictors.
+
+```{r}
+kmeans_spec <- k_means(num_clusters = 3) %>%
+  set_engine("stats")
+
+kmeans_fit <- kmeans_spec %>%
+  fit(~ ., data = ames)
+```
+
+when extracting the cluster means, we see that the dummy variables were used when calculating the means, which can lead to harder interpretations.
+
+```{r}
+kmeans_fit %>%
+  extract_centroids() %>%
+  select(101:112) %>%
+  glimpse()
+```
+
+Fitting a K-prototype model is done by setting the engine in `k_means()` to `"clustMixType"`.
+
+```{r}
+kproto_spec <- k_means(num_clusters = 3) %>%
+  set_engine("clustMixType")
+
+kproto_fit <- kproto_spec %>%
+  fit(~ ., data = ames)
+```
+
+the clusters can now be extracted on the original data format as categorical predictors are supported.
+
+```{r}
+kproto_fit %>%
+  extract_centroids() %>%
+  select(11:20) %>%
+  glimpse()
+```
+
+## Stricter rsample functions
+
+Before version 1.2.0 of rsample, misspelled and wrongly used arguments would be swallowed silently by the functions. This could be a big source of confusion as it is easy to slip between the cracks. We have many changes to all rsample functions such that whenever possible they alert the user when something is wrong. 
```

Comment:
```suggestion
Before version 1.2.0 of rsample, misspelled and wrongly used arguments would be swallowed silently by the functions. This could be a big source of confusion as it is easy to slip between the cracks. We have made changes to all rsample functions such that whenever possible they alert the user when something is wrong. 
```